wizeline/falcon/middlewares/json.py

```diff
@@ -1,52 +1,54 @@
 import json
-from json.decoder import JSONDecodeError
+from json import JSONDecodeError
 
-import falcon
+from falcon import (
+    HTTPUnsupportedMediaType,
+    HTTPBadRequest,
+    HTTPInternalServerError
+)
 
 
 class JSONMiddleware:
     def process_resource(self, req, resp, resource, params):
-        if self._is_middleware_enabled(resource) and self._request_method_has_payload(req):
-            if not self._is_json_content_type(req):
-                raise falcon.HTTPUnsupportedMediaType()
+        if (self._is_middleware_enabled(resource)
+           and self._has_request_method_payload(req)):
+            if not self._is_content_type_valid(req):
+                raise HTTPUnsupportedMediaType()
 
             try:
                 req.text = self._get_payload(req)
+                req.json = (json.loads(req.text, encoding='utf-8')
+                            if req.text.strip() != '' else {})
+            except JSONDecodeError as error:
+                raise HTTPBadRequest(f'Invalid JSON received: error={error}')
```

Comment:
Should we also log the error?

Reply:
Since this is a public library. We don't want start to add logs here. Instead, we should provide a logging mechanism which is configurable so that, users of this can define logging purposes as they want. However, this effort is out of the scope of this fix.